### ISSUE TITLE
EF.Utilities doesn't check for Constraints

### ISSUE THREAD

**Issue by pjganesh** [MikaelEliasson/EntityFramework.Utilities#77]
Hi,

There are no options available in EntityFramework.Utilities to check for constraints while doing Batch Insert operation.

Can you please check and provide a work around for the same.

Thanks in advance.

**Comment by abatishchev** [comment]
Please. +1

**Comment by TheDevKnight** [comment]
Please add this feature +1

**Comment by abatishchev** [comment]
Any update on this? Bulk insert is completely broken without it.

**Comment by wind-raven** [comment]
InsertAll supports checking of all constraints (primary / foreign keys, unique constraints ETC) since the SQL engine checks those on all inserts (including bulk insert).  DML Triggers, however, can be bypassed when using a bulk insert and require speci.  Is that the issue you are encountering?

That can be fixed by adding a flag for FIRE_TRIGGERS to be added to the options on bulk insert.

**Comment by abatishchev** [comment]
See https://github.com/MikaelEliasson/EntityFramework.Utilities/issues/34

**Comment by TheDevKnight** [comment]
@wind-raven this is not true, i could save a row with a foreignkey who points on a invalid row in the target table. InsertAll will not throw an exception or anything else :(

**Comment by abatishchev** [comment]
Exactly, that's why we're using EF.BulkInsert instead which doesn't support bulk delete.

**Comment by abatishchev** [comment]
Any update on this, please?
